VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -53,14 +56,30 @@ public OauthRefreshTokenAuthenticator(Context context) {
     public Request authenticate(Route route, final Response response) throws IOException {
         logger.warn(response.toString());
 
-        if (!isTokenExpired(response.peekBody(HttpStatus.OK).string())) {
-            return null;
-        }
-
         final AuthResponse currentAuth = loginPrefs.getCurrentAuth();
         if (null == currentAuth || null == currentAuth.refresh_token) {
             return null;
         }
+        String response_body = response.peekBody(200).string();
+
+        switch (getErrorCode(response_body)) {
+            case TOKEN_EXPIRED_ERROR_MESSAGE:
+                break;
```

Comment:
Why would the `default` case matter here? It just returns `null` in that case, right? I think the compiler would recognize that all the cases return something, but in any case you could move the default case out of the `switch` if you wanted. Anyway, it's a small change, you can do it however you like.